File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

N= int(input())
operations = [list(input().split()) for _ in range(N)]

def solution(operations):
    stack = []
    for op in operations:
        result = -1
        if op[0] == 'push':
            stack.append(int(op[1]))
            continue
        elif op[0] == 'pop':
            if stack:
                result = stack.pop()
        elif op[0] == 'size':
            result = len(stack)
        elif op[0] == 'empty':
            result = 1 if not stack else 0
        else:
            if stack:
                result = stack[-1]
        print(result)

solution(operations)

File: 백준/Silver/10828. 스택/스택.js
const fs = require("fs");
let [_, ...ops] = fs
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./input.txt")
  .toString()
  .trim()
  .split("\n");

let s = [];
const out = [];

for (const line of ops) {
  const [cmd, val] = line.split(" ");
  switch (cmd) {
    case "push":
      s.push(val);
      break;

    case "pop":
      out.push(s.length ? s.pop() : -1);
      break;

    case "size":
      out.push(s.length);
      break;

    case "empty":
      out.push(s.length ? 0 : 1);
      break;

    case "top":
      out.push(s.length ? s[s.length - 1] : -1);
      break;
  }
}

console.log(out.join("\n"));
